api/spec/controllers/spree/api/orders_controller_spec.rb

```diff
@@ -83,40 +73,41 @@ module Spree
       let(:can_admin) { false }
       subject { api_put :update, id: order.to_param, order: order_params }
 
-      before do
-        allow_any_instance_of(Spree::Ability).to receive(:can?).
-          and_return(true)
+      context "when the user cannot administer the order" do
+        stub_authorization! do |_|
+          can [:update], Spree::Order
+        end
 
-        allow(Spree::Order).to receive(:find_by!).
-          with(number: order.number).
-          and_return(order)
+        it "updates the cart contents" do
```

Comment:
The spec checks for the order's email being changed, not for the cart contents. Huh?